src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs

```diff
@@ -21,7 +24,41 @@ public static void SetDllName(string dllName)
             _dllName = dllName;
         }
 
-        private static ISqlite3 Sqlite3 => _sqlite3.Value;
+        private static ISqlite3 Sqlite3
+        {
+            get
+            {
+                if (!_sqlite3.IsValueCreated)
+                {
+                    OnLoad();
+                }
+                return _sqlite3.Value;
+            }
+        }
+
+        static partial void OnLoad();
+
+#if !NET451
+        private static bool _isLoaded;
+
+        static partial void OnLoad()
+        {
+            if (_isLoaded)
+            {
+                // prevent re-entrant calls
+                return;
+            }
+            _isLoaded = true;
+            if (ApplicationDataHelper.TemporaryFolderPath != null)
+            {
+                using (var connection = new SqliteConnection("Data Source=:memory:"))
+                {
+                    connection.Open();
+                    connection.ExecuteNonQuery("PRAGMA temp_store_directory = '" + ApplicationDataHelper.TemporaryFolderPath + "';");
+                }
```

Comment:
The more I think about this, the more I think we should make native calls directly to do all of this.